src/panels/lovelace/special-rows/hui-conditional-row.ts

```diff
@@ -0,0 +1,67 @@
+import { createRowElement } from "../common/create-row-element";
+import {
+  checkConditionsMet,
+  validateConditionalConfig,
+} from "../../lovelace/common/validate-condition";
+import { HomeAssistant } from "../../../types";
+import { LovelaceRow, ConditionalRowConfig } from "../entity-rows/types";
+
+class HuiConditionalRow extends HTMLElement implements LovelaceRow {
+  private _hass?: HomeAssistant;
+  private _config?: ConditionalRowConfig;
+  private _row?: LovelaceRow;
+
+  public setConfig(config) {
+    if (
+      !config.row ||
+      !config.conditions ||
+      !Array.isArray(config.conditions) ||
+      !validateConditionalConfig(config.conditions)
+    ) {
+      throw new Error("Error in row configuration.");
```

Comment:
We might want to be more specific in our errors, you can search a long time for what you did wrong now.